src/reformat_libyuv.c

```diff
@@ -350,429 +382,125 @@ avifResult avifImageYUVToRGBLibYUV8bpc(const avifImage * image,
-        // AVIF_RGB_FORMAT_RGB   *ToRGB24Matrix  matrixYVU
-
-        if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {
-            if (I420ToRGB24Matrix(image->yuvPlanes[AVIF_CHAN_Y],
-                                  image->yuvRowBytes[AVIF_CHAN_Y],
-                                  image->yuvPlanes[AVIF_CHAN_V],
-                                  image->yuvRowBytes[AVIF_CHAN_V],
-                                  image->yuvPlanes[AVIF_CHAN_U],
-                                  image->yuvRowBytes[AVIF_CHAN_U],
-                                  rgb->pixels,
-                                  rgb->rowBytes,
-                                  matrixYVU,
-                                  image->width,
-                                  image->height) != 0) {
-                return AVIF_RESULT_REFORMAT_FAILED;
-            }
-            return AVIF_RESULT_OK;
-        }
-    } else if (rgb->format == AVIF_RGB_FORMAT_BGR) {
-        // AVIF_RGB_FORMAT_BGR   *ToRGB24Matrix  matrixYUV
+    // AVIF_RGB_FORMAT_RGB_565   *ToRGB565Matrix  matrixYUV
 
-        if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {
-            if (I420ToRGB24Matrix(image->yuvPlanes[AVIF_CHAN_Y],
+    // Look up table for YUV To RGB Matrix (without filter).
+    typedef int (
+        *YUVToRGBMatrix)(const uint8_t *, int, const uint8_t *, int, const uint8_t *, int, uint8_t *, int, const struct YuvConstants *, int, int);
+    YUVToRGBMatrix lutYuvToRgbMatrix[RGB_PIXEL_FORMATS][YUV_PIXEL_FORMATS] = {
+        { NULL, NULL, NULL, I420ToRGB24Matrix, NULL },                        // RGB
+        { NULL, I444ToARGBMatrix, I422ToARGBMatrix, I420ToARGBMatrix, NULL }, // RGBA
+        { NULL, NULL, I422ToRGBAMatrix, I420ToRGBAMatrix, NULL },             // ARGB
+        { NULL, NULL, NULL, I420ToRGB24Matrix, NULL },                        // BGR
+        { NULL, I444ToARGBMatrix, I422ToARGBMatrix, I420ToARGBMatrix, NULL }, // BGRA
+        { NULL, NULL, I422ToRGBAMatrix, I420ToRGBAMatrix, NULL },             // ABGR
+        { NULL, NULL, I422ToRGB565Matrix, I420ToRGB565Matrix, NULL },         // RGB_565
+    };
+
+    // Look up table for YUV To RGB Matrix (with filter).
+    typedef int (*YUVToRGBMatrixFilter)(const uint8_t *,
+                                        int,
+                                        const uint8_t *,
+                                        int,
+                                        const uint8_t *,
+                                        int,
+                                        uint8_t *,
+                                        int,
+                                        const struct YuvConstants *,
+                                        int,
+                                        int,
+                                        enum FilterMode);
+    YUVToRGBMatrixFilter lutYuvToRgbMatrixFilter[RGB_PIXEL_FORMATS][YUV_PIXEL_FORMATS] = {
+        {},                                                                   // RGB
```

Comment:
UPDATE: I found that the AppVeyor build failed because of this:

```
C:\projects\libavif-93od3\src\reformat_libyuv.c(427,10): error C2059: syntax error: '}' [C:\projects\libavif-93od3\build\avif.vcxproj]
C:\projects\libavif-93od3\src\reformat_libyuv.c(538,10): error C2059: syntax error: '}' [C:\projects\libavif-93od3\build\avif.vcxproj]
C:\projects\libavif-93od3\src\reformat_libyuv.c(561,10): error C2059: syntax error: '}' [C:\projects\libavif-93od3\build\avif.vcxproj]
```